I see `-Wdeprecated-copy` warnings building just `bitcoind` with LLVM Clang 10.0.0 on macOS i.e:
```bash
make src/bitcoind -j8
...
In file included from addrman.cpp:6:
In file included from ./addrman.h:15:
In file included from ./util/system.h:20:
In file included from ./fs.h:14:
In file included from /usr/local/include/boost/filesystem.hpp:16:
In file included from /usr/local/include/boost/filesystem/path.hpp:32:
/usr/local/include/boost/io/detail/quoted_manip.hpp:62:23: warning: definition of implicit copy constructor for 'quoted_proxy<const std::__1::basic_string<char> &, char>' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
        quoted_proxy& operator=(const quoted_proxy&);  // = deleted
                      ^
/usr/local/include/boost/io/detail/quoted_manip.hpp:166:14: note: in implicit copy constructor for 'boost::io::detail::quoted_proxy<const std::__1::basic_string<char> &, char>' first required here
      return detail::quoted_proxy<std::basic_string<Char, Traits, Alloc> const &, Char>
             ^
/usr/local/include/boost/filesystem/path.hpp:827:21: note: in instantiation of function template specialization 'boost::io::quoted<char, std::__1::char_traits<char>, std::__1::allocator<char> >' requested here
      << boost::io::quoted(p.template string<std::basic_string<Char> >(), static_cast<Char>('&'));
                    ^
./tinyformat.h:358:13: note: in instantiation of function template specialization 'boost::filesystem::operator<<<char, std::__1::char_traits<char> >' requested here
        out << value;
            ^
./tinyformat.h:543:13: note: in instantiation of function template specialization 'tinyformat::formatValue<boost::filesystem::path>' requested here
            formatValue(out, fmtBegin, fmtEnd, ntrunc, *static_cast<const T*>(value));
            ^
./tinyformat.h:519:27: note: in instantiation of function template specialization 'tinyformat::detail::FormatArg::formatImpl<boost::filesystem::path>' requested here
            m_formatImpl(&formatImpl<T>),
                          ^
./tinyformat.h:975:32: note: (skipping 1 context in backtrace; use -ftemplate-backtrace-limit=0 to see all)
            m_formatterStore { FormatArg(args)... }
                               ^
./tinyformat.h:1028:12: note: in instantiation of function template specialization 'tinyformat::detail::FormatListN<2>::FormatListN<boost::filesystem::path, int>' requested here
    return detail::FormatListN<sizeof...(args)>(args...);
           ^
./tinyformat.h:1064:23: note: in instantiation of function template specialization 'tinyformat::makeFormatList<boost::filesystem::path, int>' requested here
    vformat(out, fmt, makeFormatList(args...));
                      ^
./tinyformat.h:1073:5: note: in instantiation of function template specialization 'tinyformat::format<boost::filesystem::path, int>' requested here
    format(oss, fmt, args...);
    ^
./logging.h:169:28: note: in instantiation of function template specialization 'tinyformat::format<boost::filesystem::path, int>' requested here
            log_msg = tfm::format(fmt, args...);
                           ^
addrman.cpp:638:5: note: in instantiation of function template specialization 'LogPrintf<boost::filesystem::path, int>' requested here
    LogPrintf("Opened asmap file %s (%d bytes) from disk\n", path, length);
    ^
1 warning generated.
```

So if we are going to suppress these I don't think it should be qt specific.

[`-Wdeprecated-copy` warnings](https://gist.github.com/fanquake/f822aaa1d9359182a8ad9c11d0c401e0).

--- a/configure.ac
+++ b/configure.ac
@@ -355,6 +355,7 @@ if test "x$CXXFLAGS_overridden" = "xno"; then
   AX_CHECK_COMPILE_FLAG([-Wunused-local-typedef],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-unused-local-typedef"],,[[$CXXFLAG_WERROR]])
   AX_CHECK_COMPILE_FLAG([-Wdeprecated-register],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-deprecated-register"],,[[$CXXFLAG_WERROR]])
   AX_CHECK_COMPILE_FLAG([-Wimplicit-fallthrough],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-implicit-fallthrough"],,[[$CXXFLAG_WERROR]])
+  AX_CHECK_COMPILE_FLAG([-Wdeprecated-copy],[NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-deprecated-copy"],,[[$CXXFLAG_WERROR]])
 fi
 
 enable_sse42=no

